gui/src/components/HomeComponents/CreateOrEditProjectBody.tsx

```diff
@@ -101,11 +173,36 @@ export default function CreateOrEditProjectBody({
         framework: selectedBackendFramework,
         frontend_framework: selectedFrontendFramework,
         description: projectDescription,
+        repository: useExternalGit ? selectedRepository.label : undefined,
+        repository_url: useExternalGit ? selectedRepository.value : undefined,
       };
       await toCreateNewProject(newProjectPayload);
     }
   };
 
+  useEffect(() => {
+    (async function () {
```

Comment:
can you please move the async function outside the useEffect and call it inside the useEffect this will ensure better code readability